ponos/pkg/aggregator/aggregator.go

```
@@ -284,13 +271,42 @@ func (a *Aggregator) registerAvs(avs *aggregatorConfig.AggregatorAvs) error {
 		return fmt.Errorf("AVS Execution Manager for %s already exists", avs.Address)
 	}
 
-	// Create AvsExecutionManagerInfo and add to map
+	avsCtx, avsCancel := context.WithCancel(ctx)
+
 	avsInfo := &AvsExecutionManagerInfo{
 		Address:          avs.Address,
 		ExecutionManager: aem,
-		CancelFunc:       nil, // Will be set when started
+		CancelFunc:       avsCancel,
 	}
 	a.avsManagers[avs.Address] = avsInfo
+	a.avsMutex.Unlock()
+
+	a.logger.Sugar().Infow("Starting AVS execution manager",
+		zap.String("avsAddress", avs.Address))
+
+	if err := aem.Init(avsCtx); err != nil {
+		a.avsMutex.Lock()
+		delete(a.avsManagers, avs.Address)
+		a.avsMutex.Unlock()
+		avsCancel()
+		return fmt.Errorf("failed to initialize AVS Execution Manager for %s: %w", avs.Address, err)
+	}
+
```

Once we have the Aggregator with the root context, you can spawn a new child context from it with a timeout of 500 milliseconds here. 

Create a startErr channel. When the go routine you spawn finishes the invocation of `aem.Start()` you can return the error into that channel and return that to the caller. The switch statement will timeout waiting for the result of `Start` if the context reaches its timeout of 500ms and is `Done`.